runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/HttpWatcherTask.java

```diff
@@ -0,0 +1,212 @@
+
+    //If server does not support long-polling use this interval
+    private final int pollIntervalSeconds;
+    private volatile boolean closed = false;
+
+    public HttpWatcherTask(
+        BiFunction<URL, String, NamespaceConfig> changeListener,
+        int pollIntervalSeconds)
+    {
+        super(changeListener);
+        this.etags = new ConcurrentHashMap<>();
+        this.configHashes = new ConcurrentHashMap<>();
+        this.futures = new ConcurrentHashMap<>();
+        this.configWatcherQueue = new LinkedBlockingQueue<>();
+        this.pollIntervalSeconds = pollIntervalSeconds;
+    }
+
+    @Override
+    public Void call()
+    {
+        while (!closed)
+        {
+            if (!configWatcherQueue.isEmpty())
+            {
+                URI configURI = configWatcherQueue.poll();
+                String etag = etags.getOrDefault(configURI, "");
+                sendAsync(configURI, etag);
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void watch(
+        URL configURL)
+    {
+        URI configURI = getUri(configURL);
+        sendAsync(configURI, INTIAL_ETAG);
+    }
+
+    @Override
+    public void doInitialConfiguration(
+        URL configURL) throws Exception
+    {
+        URI configURI = getUri(configURL);
+        HttpClient client = HttpClient.newBuilder()
+            .version(HTTP_2)
+            .followRedirects(NORMAL)
+            .build();
+        HttpRequest request = HttpRequest.newBuilder()
+            .GET()
+            .headers("If-None-Match", INTIAL_ETAG, "Prefer", "wait=86400")
+            .uri(configURI)
+            .build();
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        NamespaceConfig initialConfig = handleConfigChange(response);
+        if (initialConfig == null)
+        {
+            throw new Exception("Parsing of the initial configuration failed.");
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        futures.values().forEach(future -> future.cancel(true));
+        closed = true;
+    }
+
+    private void sendAsync(
+        URI configURI,
+        String etag)
+    {
+        HttpClient client = HttpClient.newBuilder()
+            .version(HTTP_2)
+            .followRedirects(NORMAL)
+            .build();
+        HttpRequest request = HttpRequest.newBuilder()
+            .GET()
+            .headers("If-None-Match", etag, "Prefer", "wait=86400")
+            .uri(configURI)
+            .build();
+        CompletableFuture<Void> future = client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+            .thenAccept(this::handleConfigChange)
+            .exceptionally(ex -> handleException(ex, configURI));
+        futures.put(configURI, future);
+    }
+
+    private Void handleException(
+        Throwable throwable,
+        URI configURI)
+    {
+        try
+        {
+            TimeUnit.SECONDS.sleep(pollIntervalSeconds);
```

Comment:
In general, I think we need to schedule the next async send rather than sleeping synchronously.